datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -613,6 +625,14 @@ macro_rules! get_data_page_statistics {
     ($stat_type_prefix: ident, $data_type: ident, $iterator: ident) => {
         paste! {
             match $data_type {
+                Some(DataType::Boolean) => Ok(Arc::new(
+                    BooleanArray::from_iter(
+                        [<$stat_type_prefix BooleanDataPageStatsIterator>]::new($iterator)
+                            .flatten()
+                            .collect::<Vec<_>>()
```

Comment:
Maybe we could add a comment here about why the `collect` is required as it is somewhat subtle

```suggestion
                            // BooleanArray::from_iter required a sized iterator, so collect into Vec first
                            .collect::<Vec<_>>()
```

Reply:
that's true, thanks!